validate/__main__.py

```diff
@@ -24,39 +24,40 @@
 def main():
     '''Main function and CLI tool'''
     args = parse_args(sys.argv[1:])
-    path = Path(args.path)
+    paths = [Path(i) for i in args.path]
     input_type = args.type
     file_type = input_type
 
-    try:
-        path_exists(path)
-    except IOError as err:
-        sys.exit(f"Error: {args.path} {str(err)}")
-
-    try:
-        if input_type in FILE_TYPES_DICT:
-            validate_file(path, input_type)
-        elif input_type == GENERIC_FILE_TYPE:
-            file_type, _ = detect_file_type_and_ext(path)
-            validate_file(path, file_type)
-        elif input_type in DIR_TYPES:
-            validate_dir(path, input_type)
-        elif input_type in CHECKSUM_GEN_TYPES:
-            create_checksum_file(path, input_type)
-    except TypeError as err:
-        sys.exit(f"Error: {args.path} {str(err)}") # raise errors, has implicit exit code
-    except ValueError as err:
-        sys.exit(f"Error: {args.path} {str(err)}")
-    except (IOError, OSError) as err:
-        sys.exit(f"Error: {args.path} {str(err)}")
-
-    print(f"Input: {args.path} is valid {file_type}")
+    for path in paths:
+        try:
+            path_exists(path)
+        except IOError as err:
+            sys.exit(f"Error: {str(path)} {str(err)}")
+
+        try:
+            if input_type in FILE_TYPES_DICT:
+                validate_file(path, input_type)
+            elif input_type == GENERIC_FILE_TYPE:
+                file_type, _ = detect_file_type_and_ext(path)
+                validate_file(path, file_type)
+            elif input_type in DIR_TYPES:
+                validate_dir(path, input_type)
+            elif input_type in CHECKSUM_GEN_TYPES:
+                create_checksum_file(path, input_type)
+        except TypeError as err:
+            sys.exit(f"Error: {path} {str(err)}") # raise errors, has implicit exit code
+        except ValueError as err:
+            sys.exit(f"Error: {path} {str(err)}")
+        except (IOError, OSError) as err:
+            sys.exit(f"Error: {path} {str(err)}")
+
+        print(f"Input: {path} is valid {file_type}")
```

Comment:
Should the program be exiting with an error if one of the input files causes a function from lines 38-46 to throw an exception? Right now if the first input is bad but the rest of the inputs are good, it exits immediately after the first input without checking the ones that are okay. Maybe more helpful behavior would be collecting the errors/successes and waiting to exit with an error after all the inputs go through validation so that there's more info about all the files that fail and which ones are good... I'm not sure what the best practice is here.

Reply:
>Maybe more helpful behavior would be collecting the errors/successes and waiting to exit with an error after all the inputs go through validation so that there's more info about all the files that fail and which ones are good.

This makes sense to me. There would be cases where multiple inputs don't exist, for example.

Reply:
It is generally a better user a experience to collect as many errors as you can upfront. Consider websites with strict password requirements. It can be frustrating to type a password, get back an error, correct it, submit, get another error, correct, submit, error, ad infinitum. Collecting all errors up front allows us to give users more immediate and useful feedback.

Here, it may make sense to print successes as they occur, while also collecting errors, and then spitting the errors out as an "error report" of sorts at the end, giving the user even more information up front.